roles/idrac_gather_facts/tests/asserts/controller_assert.yml

```diff
@@ -10,30 +11,60 @@
     body_format: "{{ idrac_gather_facts_uri_body_format }}"
     status_code: "{{ idrac_gather_facts_uri_status_code }}"
     return_content: "{{ idrac_gather_facts_uri_return_content }}"
-  register: controller_result
+  register: storage_entity
   no_log: true
 
-- name: Set controller facts
+- name: Get storage entities
+  ansible.builtin.set_fact:
+    collected_storage_entity:
+      "{{ (collected_storage_entity | default([])) + [item['@odata.id']] }}"
+  loop: "{{ storage_entity.json.Members }}"
+
+- name: Get Controllers URL information.
+  ansible.builtin.uri:
+    url: "https://{{ hostname }}{{ item }}"
+    <<: *uri_params
+  loop: "{{ collected_storage_entity }}"
+  register: controllers_url
+
+- name: Pick Controllers key value in one list
   ansible.builtin.set_fact:
-    api_response: "{{ controller_result.json | ansible.utils.remove_keys(target=['@odata.context', '@odata.id', '@odata.type']) }}"
+    controllers_list:
+      "{{ controllers_list | default([]) +
+       [item.json.Controllers | default(omit)] }}"
+  loop: "{{ controllers_url.results }}"
 
-- name: Set the keys diff
+- name: Select list of controller values from dictionary
   ansible.builtin.set_fact:
-    diff_keys: "{{ controller_data.keys() | list | symmetric_difference((api_response.keys() | list)) }}"
+    collected_values:
+      "{{ (collected_values | default([])) +
+       [item['@odata.id'] | default(omit)] }}"
+  loop: "{{ controllers_list }}"
 
-- name: Set a diff of dict
+- name: Get Controllers information.
+  ansible.builtin.uri:
+    url: "https://{{ hostname }}{{ item }}?$expand=*($levels=1)"
+    <<: *uri_params
+  loop: "{{ collected_values | reject('match', '__omit_place_holder__.*') }}"
+  register: result
```

Comment:
can you rename this register var to idrac_gather_facts_

Reply:
Done